### PR TITLE
handle LocalizationConfusion memory consuption and upgrade min weasyprint version

### LINKED ISSUE #1042: [Bug] LocalizationConfusion metric with use_polygons (rotation=True) results in enormous memory consumption
### Bug description

Ref.: #935 

duplicate:
- with broadcasting (default) it would kill every normal machine (FUNSD evaluation needs > 64GB RAM)
- without broadcasting it is ultra slow (but constant)

```
def polygon_iou(
    polys_1: np.ndarray, polys_2: np.ndarray, mask_shape: Tuple[int, int], use_broadcasting: bool = False
) -> np.ndarray:
    """Computes the IoU between two sets of rotated bounding boxes

    Args:
        polys_1: rotated bounding boxes of shape (N, 4, 2)
        polys_2: rotated bounding boxes of shape (M, 4, 2)
        mask_shape: spatial shape of the intermediate masks
        use_broadcasting: if set to True, leverage broadcasting speedup by consuming more memory

    Returns:
        the IoU matrix of shape (N, M)
    """

    if polys_1.ndim != 3 or polys_2.ndim != 3:
        raise AssertionError("expects boxes to be in format (N, 4, 2)")

    iou_mat: np.ndarray = np.zeros((polys_1.shape[0], polys_2.shape[0]), dtype=np.float32)

    if polys_1.shape[0] > 0 and polys_2.shape[0] > 0:
        if use_broadcasting:
            masks_1 = rbox_to_mask(polys_1, shape=mask_shape)
            masks_2 = rbox_to_mask(polys_2, shape=mask_shape)
            iou_mat = mask_iou(masks_1, masks_2)
        else:
            # Save memory by doing the computation for each pair
            for idx, b1 in enumerate(polys_1):
                m1 = _rbox_to_mask(b1, mask_shape)
                for _idx, b2 in enumerate(polys_2):
                    m2 = _rbox_to_mask(b2, mask_shape)
                    iou_mat[idx, _idx] = np.logical_and(m1, m2).sum() / np.logical_or(m1, m2).sum()

    return iou_mat
```

### Code snippet to reproduce the bug

python3  /home/felix/Desktop/doctr/references/detection/evaluate_pytorch.py db_mobilenet_v3_large --rotation

### PR DESCRIPTION
**Pull request by felixdittrich92**
This PR:

- update references to avoid that machines with less memory will be killed while computing the metric (workaround to close #1042 )
- bump weasyprint to >=55.0 (html to pdf bugfixes >=54.0 new pango version)

Any feedback is welcome :hugs:

### CODE REVIEW

File: references/detection/evaluate_pytorch.py [hunk 1]
```diff
@@ -130,7 +131,11 @@ def main(args):
         model = model.cuda()
 
     # Metrics
-    metric = LocalizationConfusion(use_polygons=args.rotation, mask_shape=input_shape)
+    metric = LocalizationConfusion(
+        use_polygons=args.rotation,
+        mask_shape=input_shape,
+        use_broadcasting=True if int(psutil.virtual_memory().total / 1024**3) > 63 else False,
```
Comment (frgfm) [inline references/detection/evaluate_pytorch.py:137]: I think that's a dangerous "non-human" conditional arg

**Review by frgfm — COMMENTED** [review]
Thanks! cf my comment :)

File: references/detection/evaluate_pytorch.py (hunk 1 above)
Comment (felixdittrich92) [reply references/detection/evaluate_pytorch.py:137]: yeah i agree :sweat_smile: will make it more readable

**Comment by felixdittrich92** [comment]
failing test related to #1063 (will work after this is merged - i have updated the dummy model on hf hub)

**Comment by felixdittrich92** [comment]
@frgfm now better ? :)

File: references/detection/evaluate_pytorch.py [hunk 2]
```diff
@@ -130,7 +132,11 @@ def main(args):
         model = model.cuda()
 
     # Metrics
-    metric = LocalizationConfusion(use_polygons=args.rotation, mask_shape=input_shape)
+    metric = LocalizationConfusion(
+        use_polygons=args.rotation,
+        mask_shape=input_shape,
+        use_broadcasting=True if system_total_memory > 62 else False,
```
[the remark this answers is not in the document]
Comment (felixdittrich92) [reply references/detection/evaluate_pytorch.py:138]: Yep the first computation eats your machine with broadcasting:sweat_smile:

File: references/detection/evaluate_pytorch.py [hunk 3]
```diff
@@ -63,6 +64,7 @@ def main(args):
         args.workers = min(16, mp.cpu_count())
 
     torch.backends.cudnn.benchmark = True
+    system_total_memory = int(psutil.virtual_memory().total / 1024**3)
```
[the remark this answers is not in the document]
Comment (felixdittrich92) [reply references/detection/evaluate_pytorch.py:67]: Good catch yep will change this tomorrow 👍

**Review by odulcy-mindee — APPROVED** [review]
Thanks ! :rocket:

**Pull request merged**